pymeasure/adapters/prologix.py

```diff
@@ -103,6 +103,49 @@ def write(self, command):
         command += "\n"
         self.connection.write(command.encode())
 
+    def _format_binary_values(self, values, datatype='f', is_big_endian=False):
+        """Format values in binary format in order to be used with instrument commands.
+
+        :param values: data to be writen to the device.
+        :param datatype: the format string for a single element. See struct module.
+        :param is_big_endian: boolean indicating endianess.
+        :return: binary string.
+        :rtype: bytes
+        """
+
+        block = super()._format_binary_values(values, datatype, is_big_endian)
+        # Prologix needs certian characters to be escaped.
+        # Special care must be taken when sending binary data to instruments. If any of the
+        # following characters occur in the binary data -- CR (ASCII 13), LF (ASCII 10), ESC
+        # (ASCII 27), '+' (ASCII 43) - they must be escaped by preceding them with an ESC
+        # character.
+        special_chars = b'\x0d\x0a\x1b\x2b'
+        new_block = b''
+        for b in block:
+            escape = b''
+            if b in special_chars:
+                escape = b'\x1b'
+            new_block += (escape + bytes((b,)))
+
+        return new_block
+
+    def write_binary_values(self, command, values, **kwargs):
+        """ Write binary data to the instrument, e.g. waveform for signal generators.
+
+        values are encoded in a binary format according to
+        IEEE 488.2 Definite Length Arbitrary Block Response Data block.
+
+        :param command: SCPI command to be sent to the instrument
+        :param values: iterable representing the binary values
+        :param kwargs: Key-word arguments to pass onto `_format_binary_values`
```

Comment:
If you change this to 
```
:meth:`~._format_binary_values`
```
the rendered documentation should get a link to _format_binary_values so that users can find out what kwargs are actually allowed.
I don't know if a private method gets documented by default -- maybe you can check that if you are set up for this? The documentation can be generated with sphinx by running `make html` in the docs/ directory, you need the sphinx and sphinx_rtd_theme packages installed.

Reply:
I have the documentation setup, but apparently internal API does not get documented in my test.

Reply:
I have pushed a doc fix to your PR branch. :-)